bin/cron/deploy_to_test

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env ruby
+
+require 'cdo/chat_client'
+require 'cdo/github'
+require 'cdo/only_one'
+
+# Checks whether the topic of Slack#developers includes "DTT: yes", updating it
+# to reflect auto-DTT if so.
+# @return [Boolean] Whether to proceed with the auto-DTT.
+def check_and_update_developers_topic
+  current_topic = Slack.get_topic('developers')
+  unless current_topic.include? 'DTT: yes'
+    ChatClient.message(
+      'infra-test',
+      "auto-DTT aborted (\#developers topic is #{current_topic}).",
+      color: 'yellow'
+    )
+    return false
+  end
+
+  new_topic = current_topic.gsub('DTT: yes', 'DTT: no (robo-DTT in progress)')
+  Slack.update_topic(new_topic)
+
+  # TODO(asher): Determine if we should continue with the DTT if the Slack topic
+  # update failed.
+  true
+end
+
+def main
+  return unless check_and_update_developers_topic
+
+  pr_number = GitHub.create_pull_request(
+    title: 'DTT (Staging > Test) [robo-dtt]',
+    head: 'staging',
+    base: 'test'
+  )
+  if pr_number.nil?
+    ChatClient.message(
+      'infra-test',
+      'GitHub.create_pull request failed.',
+      color: 'red'
+    )
+    return
```

Comment:
In this failure case you're not setting the developers' topic back to `DTT: yes`.  Actually, this might be a great place for a rescue/ensure structure; raise if anything goes wrong, report problems in rescue blocks, and reset the developers topic in the ensure block.